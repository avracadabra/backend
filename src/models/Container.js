import { Model } from "@vuex-orm/core";
import ContainerType from "@/models/ContainerType";

export default class Container extends Model {
  // This is the name used as module name of the Vuex Store.
  static entity = "containers";
  // List of all fields (schema) of the post model. `this.attr` is used
  // for the generic field type. The argument is the default value.
  static fields() {
    return {
      id: this.number(null).nullable(),
      code: this.string(""),
      type_id: this.number(null).nullable(),
      type: this.belongsTo(ContainerType, "type_id"),
    };
  }
}
