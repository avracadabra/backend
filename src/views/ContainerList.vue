<template>
  <div id="container-list-page">
    <b-table id="container-list" :data="containers" :columns="columns">
      <template slot="empty">
        <section class="section">
          <div class="content has-text-grey has-text-centered">
            <p>
              <b-icon icon="emoticon-sad" size="is-large"> </b-icon>
            </p>
            <p>No containers found with those criteria.</p>
          </div>
        </section>
      </template>
    </b-table>
  </div>
</template>
<script>
import Container from "@/models/Container";
// import { ContainerType } from "@/models/containerType";

export default {
  name: "ContainerList",
  data() {
    return {
      columns: [
        {
          field: "code",
          label: "Container code",
        },
        {
          field: "type.code",
          label: "Container type code",
        },
        {
          field: "type.label",
          label: "Container type",
        },
      ],
    };
  },
  computed: {
    containers: () =>
      Container.query()
        .withAll()
        .all(),
  },
  async mounted() {
    await Container.fetch();
  },
};
</script>
