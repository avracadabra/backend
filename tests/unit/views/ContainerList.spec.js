import { mount } from "@vue/test-utils";
import ContainerList from "@/views/ContainerList";
import Container from "@/models/Container";
import { mock, clearORMStore } from "@vuex-orm/plugin-graphql";
import flushPromises from "flush-promises";

const localVue = global.localVue;
const store = global.store;

describe("ContainerList", () => {
  mock("fetch")
    .for(Container)
    .andReturn([
      {
        id: 2,
        code: "empl 1",
        type: {
          id: 3,
          code: "LOCATION",
          label: "Location",
          __typename: "ContainerType",
        },
        __typename: "Container",
      },
      {
        id: 1,
        code: "warehouse",
        type: {
          id: 3,
          code: "LOCATION",
          label: "Location",
          __typename: "ContainerType",
        },
        __typename: "Container",
      },
    ]);

  beforeEach(async () => {
    await clearORMStore();
  });

  it("mount ContainerList view", async () => {
    const wrapper = mount(ContainerList, {
      store,
      localVue,
    });
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  });
});
