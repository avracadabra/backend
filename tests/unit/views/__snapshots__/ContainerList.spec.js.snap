// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerList mount ContainerList view 1`] = `
<div
  id="container-list-page"
>
  <div
    class="b-table"
    id="container-list"
  >
    <!---->
     
    <!---->
     
    <div
      class="table-wrapper has-mobile-cards"
    >
      <table
        class="table"
      >
        <thead>
          <tr>
            <!---->
             
            <!---->
             
            <th
              class="is-unselectable"
            >
              <div
                class="th-wrap"
              >
                Container code 
                <!---->
              </div>
            </th>
            <th
              class="is-unselectable"
            >
              <div
                class="th-wrap"
              >
                Container type code 
                <!---->
              </div>
            </th>
            <th
              class="is-unselectable"
            >
              <div
                class="th-wrap"
              >
                Container type 
                <!---->
              </div>
            </th>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
        </thead>
         
        <tbody>
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Container code"
            >
              
                                        warehouse
                                    
            </td>
            <td
              class=""
              data-label="Container type code"
            >
              
                                        LOCATION
                                    
            </td>
            <td
              class=""
              data-label="Container type"
            >
              
                                        Location
                                    
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Container code"
            >
              
                                        empl 1
                                    
            </td>
            <td
              class=""
              data-label="Container type code"
            >
              
                                        LOCATION
                                    
            </td>
            <td
              class=""
              data-label="Container type"
            >
              
                                        Location
                                    
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
        </tbody>
         
        <!---->
      </table>
    </div>
     
    <!---->
  </div>
</div>
`;
